web/src/components/citations.tsx

```diff
@@ -79,25 +73,44 @@ export const ShowCitation: React.FC<{citation: Citation, i: number}> = ({citatio
   // if we don't have a url, link to a duckduckgo search for the title instead
   const url = citation.url && citation.url !== ""
         ? citation.url
-        : `https://duckduckgo.com/?q=${encodeURIComponent(citation.title)}`;
+        : `https://duckduckgo.com/?q=${encodeURIComponent(ndecitation.title)}`;
 
   return (
-    <A className={Colours[i % Colours.length] + " border-2 flex items-center rounded my-2 text-sm no-underline w-fit"}
+    <A className={Colours[(citation.index - 1) % Colours.length] + " border-2 flex items-center rounded my-2 text-sm no-underline w-fit"}
       href={url}>
-      <span className="mx-1"> [{i + 1}] </span>
+      <span className="mx-1"> [{citation.index}] </span>
       <p className="mx-1 my-0"> {c_str} </p>
     </A>
   );
 };
 
-export const ShowInTextCitation: React.FC<{citation: Citation, i: number}> = ({citation, i}) => {
-  const url = citation.url && citation.url !== ""
-        ? citation.url
-        : `https://duckduckgo.com/?q=${encodeURIComponent(citation.title)}`;
-  return (
-    <A className={Colours[i % Colours.length] + " border-2 rounded text-sm no-underline w-min px-0.5 pb-0.5 ml-1 mr-0.5"}
-      href={url}>
-      [{i + 1}]
-    </A>
-  );
+export const CitationRef: React.FC<{citation: Citation}> = ({citation}) => {
+    const url = citation.url && citation.url !== ""
+              ? citation.url
+              : `https://duckduckgo.com/?q=${encodeURIComponent(citation.title)}`;
+    return (
+        <A className={Colours[(citation.index - 1) % Colours.length] + " border-2 rounded text-sm no-underline w-min px-0.5 pb-0.5 ml-1 mr-0.5"}
+           href={url}>
+            [{citation.index}]
+        </A>
+    );
 };
+
+
+export const CitationsBlock: React.FC<{text: string, citations: Map<string, Citation>, textRenderer: (t: str) => any}> = ({text, citations, textRenderer}) => {
```

Comment:
(minor) is `React.FC` type used for consistency with the rest of the repo? just typing the props should be enough for function components, avoiding problems with implicit children - https://stackoverflow.com/a/59991281/1176601

Reply:
yes